workflows/ingest/incremental_ingest.go

```diff
@@ -83,5 +102,104 @@ func Incremental(ctx workflow.Context, params IncrementalParams) error {
 		return err
 	}
 
+	// Wait for all reaper files to be available
+	waitForFileResult := map[paths.Path]workflow.Future{}
+	for _, file := range reaperResult.Files {
+		path := paths.MustParse(file)
+		r := wfutils.ExecuteWithQueue(ctx, activities.WaitForFile, activities.FileInput{
+			Path: path,
+		})
+
+		waitForFileResult[path] = r
+	}
+
+	outputFolder, err := wfutils.GetWorkflowRawOutputFolder(ctx)
+	if err != nil {
+		return err
+	}
+
+	addSilenceFutures := map[paths.Path]workflow.Future{}
+
+	baseFileName := strings.TrimSuffix(rawPath.Base(), "_MU1.mxf")
+
+	// As Reaper files become available, prepend silence to them
+	for len(waitForFileResult) > 0 {
+		newWaitForFileResult := map[paths.Path]workflow.Future{}
+		for path, res := range waitForFileResult {
+			if !res.IsReady() {
+				newWaitForFileResult[path] = res
+				continue
+			}
+
```

Comment:
You need to retrieve the keys of the map before iterating (wfutils.GetMapKeysSafely), and only iterate through arrays (maps aren't deterministic).

Reply:
Hmmm, I'm wondering if this can be made deterministic at all since I update stuff in the loop. I think the best is to actually package the flow for each file into it's own workflow and await them.

Reply:
Oh true. should probably avoid .IsReady().

Selectors?